### DISCUSSION TITLE
AutoSquaring problem

### DISCUSSION THREAD

**Discussion by SimonLane** [phil-barrett/grblHAL-teensy-4.x#106] [Q&A] [answered]
Hi Phil, 

I'm really enjoying the breakout board and grblHAL for my 3 axis machine. I have come unstuck with the autosquare routine though. My machine works just fine with a 3XGY firmware. But recently I wanted to take advantage of the autosquare feature and added another endstop on my second Y axis (wired on 'A'), and changed to a 3XGYAS firmware. The endstop works and the red Y 'light' shows up in iosender when either endstop is triggered. The machine still works fine, apart from the homing routine. It seems to seek out both endstops, but then makes a very rapid move and I get a fault on my motor driver and a 'hard limit' alarm in iosender. It doesn't do anything like that on the others axes, or when I use the 3XGY firmware.

Any guesses what's going on or how I might test this problem to get more info?

Best Wishes,

Simon

**Comment by phil-barrett** [comment, marked as answer]
One possibility, you have the Y1 and Y2 end stops switched.

**Reply by SimonLane** [reply]
Thanks Phil,
I had double checked the switches and Y2 was indeed wired to A. My mistake was that the stepper outputs for 'Y' went to Y2, whilst the stepper outputs for 'A' went to Y1. So yes, basically my switches were backwards! Thanks for your help, much appreciated!

**Reply by phil-barrett** [reply]
Great!  The reason why I suggested that was because, er, um, I did the same thing.
